src/CarbonAware.CLI/src/Commands/Emissions/EmissionForecastCommand.cs

```diff
@@ -0,0 +1,100 @@
+﻿using CarbonAware.Aggregators.CarbonAware;
+using CarbonAware.CLI.Common;
+using CarbonAware.CLI.Model;
+using CarbonAware.Tools.WattTimeClient.Model;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Text.Json;
+
+namespace CarbonAware.CLI.Commands.Emissions;
+
+class EmissionsForecastCommand : Command
+{
+    private readonly Option<string[]> _requiredLocation = CommonOptions.RequiredLocationOption;
+    
+    private readonly Option<DateTimeOffset?> _dataEndTime = new Option<DateTimeOffset?>(
+                new string[] { "--data-end-time", "-e" },
+            LocalizableStrings.DataEndAtDecsription)
+    {
+        Arity = ArgumentArity.ZeroOrOne,
+    };
+    private readonly Option<DateTimeOffset?> _dataRequestedAt = new Option<DateTimeOffset?>(
+                new string[] { "--data-requested-at", "-r" },
+            LocalizableStrings.DataRequestedAtDescription)
+    {
+        Arity = ArgumentArity.ZeroOrOne,
+    };
+    private readonly Option<DateTimeOffset?> _dataStartTime = new Option<DateTimeOffset?>(
+            new string[] { "--data-start-time", "-s" },
+            LocalizableStrings.DataStartAtDescription)
+    {
+        Arity = ArgumentArity.ZeroOrOne,
+    };
+    private readonly Option<int?> _duration = new Option<int?>(
+                new string[] { "--duration", "-d" },
+            LocalizableStrings.DurationDescription)
+    {
+        Arity = ArgumentArity.ZeroOrOne,
+    };
+
+    public EmissionsForecastCommand() : base("emissions-forecast", LocalizableStrings.EmissionsCommandDescription)
+    {
+        AddOption(_requiredLocation);
+        AddOption(_dataStartTime);
+        AddOption(_dataEndTime);
+        AddOption(_duration);
+        AddOption(_dataRequestedAt);
+
+        this.SetHandler(this.Run);
+    }
+
+    internal async Task Run(InvocationContext context)
+    {
+        // Get aggregator via DI.
+        var serviceProvider = context.BindingContext.GetService(typeof(IServiceProvider)) as IServiceProvider ?? throw new NullReferenceException("ServiceProvider not found");
```

Comment:
nit: not a particularly useful error message so I wouldn't introduce a new string

```suggestion
        var serviceProvider = context.BindingContext.GetService(typeof(IServiceProvider)) as IServiceProvider ?? throw new NullReferenceException(nameof(serviceProvider));
```